implement Docker containerization for Baby Tools Shop

4gh0rn opened the pull request:
- **Dockerfile** with Python 3.11-slim base
- **docker-compose.prod.yml** for production
- **entrypoint.sh** for automated startup
- **build.sh** for one-command deployment
- **.env** configuration support
- **Complete documentation** in README.md

4gh0rn committed:
implement containerized Django app with Docker

4gh0rn committed:
added .env support

4gh0rn committed:
added python-dotenv==1.1.1 to reqirements

PascalNehlsen commented on docker-compose.prod.yml:
```diff
@@ -0,0 +1,8 @@
+services:
+  web:
+    build: .
+    ports:
+      - "8025:8025"
+    env_file:
+      - .env
+    restart: unless-stopped
```
please run your setup without compose

PascalNehlsen commented on entrypoint.sh:
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
```
Its okay to use a shellscript here. but for a small tools shop setup you dont need it. the migration and server starting you can do in your dockerfile. I would leave it here.

PascalNehlsen commented on Dockerfile:
```diff
@@ -0,0 +1,21 @@
+FROM python:3.11-slim
```
Take the Python version described in the readme file.

PascalNehlsen commented on Dockerfile:
```diff
@@ -0,0 +1,21 @@
+# Copy requirements, entrypoint and project files
+COPY requirements.txt .
+COPY entrypoint.sh .
+COPY babyshop_app .
```
if you want to copy this three files you can do it in one step

PascalNehlsen commented on README.md:
```diff
@@ -1,37 +1,241 @@
+# Baby Tools Shop
 
-### TECHNOLOGIES
+Here you can find the Container Architecture & Deployment Guide
```
please add a table of contents

PascalNehlsen commented on README.md:
```diff
@@ -1,37 +1,241 @@
 
-### Hints
+### Prerequisites
+- Docker installed and running
```
```suggestion
- Docker
```

PascalNehlsen commented on README.md:
```diff
@@ -1,37 +1,241 @@
-### Hints
+### Prerequisites
+- Docker installed and running
+- Docker Compose installed
```
```suggestion
```

PascalNehlsen commented on README.md:
````diff
@@ -1,37 +1,241 @@
 
-##### Home Page with login
+# Access your app: http://localhost:8025
+```
````
good! Please split this into three commands and insert the comments outside the code box.

PascalNehlsen commented on README.md:
````diff
@@ -1,37 +1,241 @@
 
+# View logs
+docker-compose -f docker-compose.prod.yml logs -f
+```
````
same as above

PascalNehlsen commented on build.sh:
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+
```
do you need this script for one command?

PascalNehlsen commented on README.md:
````diff
@@ -1,37 +1,241 @@
 
-<img alt="" src="https://github.com/MET-DEV/Django-E-Commerce/blob/master/project_images/capture_20220323081044867.jpg"></img>
+### Dockerfile Structure
+```dockerfile
````
remove it and link to your existing dockerfile

PascalNehlsen commented on README.md:
```diff
@@ -1,37 +1,241 @@
+---
 
-##### Login Page
+## Container Architecture Overview
```
add this to your #Usage section

PascalNehlsen reviewed (changes requested):
Hey Uwe,

Really good setup! I think it's great that you've already thought about Compose. However, in this setup, we also want to start our app without Compose. I would suggest describing how to start this super quickly with Compose in the Quickstart. In your newly created Usage section, I would describe how to do it without Compose. 

Otherwise, I have little to criticize about the setup. My comments are more aimed at the UX in your Readme.

Have fun :)

4gh0rn committed:
refactor: improved README structure and user experience

- Add TOC and Usage section
- Split code blocks for better copy-paste
- Link files instead of inline code
- Streamline content for clarity
- implemented wsgi with gunicorn instead of runserver
- added default value for secret key in settings.py
- fixed copy command to  use one liner in Dockerfile

PascalNehlsen commented on README.md:
````diff
@@ -1,37 +1,102 @@
-### Photos
+**1. Clone and navigate to the project**
+```bash
+git clone <your-repo-url>
````
missed in last feedback, sorry! use your repo url here.

PascalNehlsen commented on README.md:
```diff
@@ -1,37 +1,102 @@
 
-##### Home Page with no login
-<img alt="" src="https://github.com/MET-DEV/Django-E-Commerce/blob/master/project_images/capture_20220323080953570.jpg"></img>
+Access your app: http://localhost:8025
```
You dont know where the user is using your repo. I would prefer:
```suggestion
Access your app: http://<your_ip>:8025
```

4gh0rn committed:
added repo url and removed the loclahost mention to use <your-ip> instead
